### ISSUE TITLE
[Discussion] elements of wiki documentation design that are missing here

### ISSUE THREAD

**Issue by kaktusus** [yorikvanhavre/FreeCAD-documentation#7]
I expect that this (the project) is a trial run for how the FreeCAD documentation should look when moved from the Wiki platform. There are many elements missing here that are worth considering whether they should be transferred here.

Not only do they improve the appearance, but they also facilitate access to other descriptions and enhance navigation style.

* Tool icons can be easily implemented. They will liven up this wall of text a bit. I know that the Gauri project encountered this problem because it was not possible to adjust the size of the graphics. However, this is not difficult, you just need to use: `<img src="image.png" width="24" height="24">`

I think it's worth considering moving the functionality of the templates:
- Template:Docnav
- Template:TutorialInfo
- Template:Macro
- Template:GuiCommand
- Template:Button
- Template:KEY
- Template:Userdocnavi
- Template:Code
- Template:WB Tools navi
- Template:MenuCommand
- Template:PropertyData
- Template:Version *(+ - )*

:eyes:

**Comment by yorikvanhavre** [comment]
There is no template functionality in markdown. We'll need to find something else...

**Comment by kaktusus** [comment]
That's why I brought up the topic.
If the interpreter of the site will accept html code maybe it is possible to prepare a similar functionality.
Of course, I do not say that it is obligatory to do it.
I just think out loud.

I can do a poll on the forum to find out the preferences of the readers of the documentation and whether they find these templates useful *(whether they use some of them)* :grey_question:
If it will be an undesirable functionality then it is a waste of time for foam-beating. :wink:

**Comment by yorikvanhavre** [comment]
> If the interpreter of the site will accept html code maybe it is possible to prepare a similar functionality.

Yes, but the goal here is to remain flexible and adaptable, so the markdown content can be parsed in a number of ways (by a website front end, by FreeCAD's help system, to be converted into epub, pdf, etc.) so the idea is to stay as close to standard markdown as possible, otherwise we're just pushing the problems further down the road.

**Comment by kaktusus** [comment]
I understand. My perceptions were different.
So the wiki has nicer documentation. I expect a wall of text.

**Comment by yorikvanhavre** [comment]
Of course, each solution has its advantages. But there are pretty big problems with the wiki:

- we don't have reliable backups. We are constantly at risk of loosing data
- we cannot produce other formats like epub or pdf
- the search is notoriously bad
- administering users is getting complicated and problematic
- there is no way to tie a version of the wiki with a release of freecad
- there is no way to automate (for ex. the API docs)

**Comment by kaktusus** [comment]
I know, I know
I'm not a wiki defender.
I only mean to be good design.
